### PR TITLE
Standardize sentinel pattern (F-ssv-spec-027)

### PR DESCRIPTION
**Pull request by julienharbulot**
Fix for:

Inconsistent Enum Sentinel Values
ssv-spec · Inconsistency · Assigned: dev4 · Effort: —
`BNRoleUnknown = math.MaxUint64` (BeaconRole) vs `RoleUnknown = -1` (RunnerRole as int32). The RunnerRole sentinel `-1` can't be properly represented as uint32 (line 50-52 in messages.go converts to uint32), potentially matching corrupted data.

Impact: Type confusion between sentinel values. Corrupted message data could match `RoleUnknown` unintentionally.

Recommendation: Standardize sentinel pattern across all enums.

### CODE REVIEW

File: types/runner_role_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	"encoding/binary"
+	"math"
+	"testing"
+)
+
+func TestRunnerRoleWireMapping(t *testing.T) {
```
Comment (olegshmuelov) [inline types/runner_role_test.go:11]: The test covers `RoleUnknown` sentinel and the `0x80000000` boundary, but doesn't exercise the `r < 0` guard in `WireUint32()` for non-sentinel negative values. Since `WireUint32` has an explicit `r < 0` branch (defense-in-depth beyond just `r == RoleUnknown`), a single test case would document that contract.

Suggestion:
```go
// Arbitrary negative role (not RoleUnknown) should also map to sentinel.
if got := RunnerRole(-42).WireUint32(); got != uint32(math.MaxUint32) {
    t.Fatalf("RunnerRole(-42).WireUint32() = %d, want %d", got, uint32(math.MaxUint32))
}
```

Comment (julienh-ssv) [reply types/runner_role_test.go:11]: Good point — added a negative (non-sentinel) role case to document the `r < 0` contract. Fixed in aa3ead15.

**Review by olegshmuelov — DISMISSED** [review]

File: types/runner_role.go [hunk 2]
```diff
@@ -12,6 +14,30 @@ const (
...
+		return runnerRoleUnknownWireUint32
+	}
+	return uint32(r)
+}
+
+// RunnerRoleFromWireUint32 converts a 4-byte uint32 MessageID role field into RunnerRole.
+// Values outside the int32 range are treated as unknown to avoid surprising negative roles.
+func RunnerRoleFromWireUint32(v uint32) RunnerRole {
+	// Any value outside the int32 range (including the RoleUnknown sentinel) is treated as unknown.
+	if v > uint32(math.MaxInt32) {
```
Comment (nkryuchkov) [inline types/runner_role.go:35]: `uint32(math.MaxUint32)` is also an unknown role, right? 
```suggestion
	if v >= uint32(math.MaxInt32) {
```

File: types/runner_role_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+	for _, role := range []RunnerRole{
+		RoleCommittee,
+		RoleProposer,
+		RoleValidatorRegistration,
+		RoleVoluntaryExit,
+		RoleAggregatorCommittee,
+	} {
+		if got := RunnerRoleFromWireUint32(role.WireUint32()); got != role {
+			t.Fatalf("round-trip(%v) = %v, want %v", role, got, role)
```
Comment (nkryuchkov) [inline types/runner_role_test.go]: can we use `testify` in this file?

File: types/runner_role.go (hunk 2 above)
Comment (julienh-ssv) [reply types/runner_role.go:35]: No here we only exclude values out of the Int32 range to avoid negative values as explained in the function's header comment. `math.MaxInt32` is still a potentially valid RunnerRole. It's a different layer of the code that will decide it.

Comment (nkryuchkov) [reply types/runner_role.go:35]: sorry, I misread it as `math.MaxUint32` and I thought `runnerRoleUnknownWireUint32` covers it

**Review by nkryuchkov — DISMISSED** [review]

**Review by ljuba-ssv — DISMISSED** [review]

**Review by MatheusFranco99 — COMMENTED** [review]
Good change! Looks good to me! :)
However, I didn't get why the .json file tests got modified (since the edge case was already the same 0xff... , and some tests don't seem related to the issue)
Can you give me a help on that?

**Comment by julienharbulot** [comment]
@MatheusFranco99 I had regenerated the fixtures after merge but you are right it wasn't necessary. I have cleaned up the git history and removed the unnecessary changes to allow for a clean merge on top of main

**Review by olegshmuelov — APPROVED** [review]

**Review by nkryuchkov — APPROVED** [review]

**Comment by julienharbulot** [comment]
Fixes https://github.com/ssvlabs/ssv-node-board/issues/866 
@MatheusFranco99 ready for your review
